phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -159,6 +161,16 @@ public IndexRebuildRegionScanner(final RegionScanner innerScanner, final Region
         }
     }
 
+    private boolean shouldRebuildOrVerify() throws IOException {
+        if(verifyType == IndexTool.IndexVerifyType.ONLY) {
+            return true;
+        }
+        byte [] rowKey = generateResultTableRowKey(scan.getTimeRange().getMax(),
```

Comment:
rather than importing the generateResultTableRowKey here, might be cleaner to have a ResultRepository "exists" overload that takes Scan, indexMaintainer, and region name. That keeps the internal details of the table from "leaking" back into the region scanner.

Reply:
or Scan, index table name, and region name. Whichever you prefer.